src/Symfony/Component/Messenger/Transport/Receiver/SingleMessageReceiver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Receiver that decorates another, but receives only 1 specific message.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental in 4.3
+ */
+class SingleMessageReceiver implements ReceiverInterface
```

Comment:
This could be marked as `@internal`, as it's a utility for the "retry" command.

Reply:
let's do it